File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n= nums.length;
        int minIdx=-1,maxIdx=-1;
        int mini=Integer.MAX_VALUE;
        int maxi = Integer.MIN_VALUE;

        for(int i=0; i<n; i++){
            if(nums[i] > maxi){
                maxi = nums[i];
                maxIdx = i;
            }
            if(nums[i] < mini){
                mini= nums[i];
                minIdx=i;
            }

        }

        int lftDel = Math.max(minIdx,maxIdx)+1;
        int rgtDel = n - Math.min(minIdx,maxIdx);
        int bthDel = Math.min(minIdx,maxIdx)+1 + n-Math.max(minIdx,maxIdx);

        return Math.min(lftDel, Math.min(rgtDel, bthDel));
        

    }
}

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.cpp
class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n = nums.size();
        int mini=INT_MAX,maxi=INT_MIN;
        int minIndx=-1,maxIndx=-1;

        for(int i=0; i<n; i++){
            if(nums[i] > maxi){
                maxi = nums[i];
                maxIndx = i;
            }
            if(nums[i] < mini){
                mini = nums[i];
                minIndx = i;
            }
        }

        int lftDel = max(minIndx,maxIndx)+1;
        int rgtDel = n- min(minIndx,maxIndx);
        int bothDel = min(minIndx,maxIndx)+1 + n- max(minIndx,maxIndx);

        return min(lftDel,min(rgtDel,bothDel));
    }
};